jupyter_server/services/contents/manager.py

```diff
@@ -128,20 +128,39 @@ def _validate_pre_save_hook(self, proposal):
             raise TraitError("pre_save_hook must be callable")
         return value
 
-    def run_pre_save_hook(self, model, path, **kwargs):
```

Comment:
Let's put a deprecation cycle on this method name too since it's a public API.

We can shim this method to the new method name, run_pre_save_hooks and raise a warning in this method.

Reply:
Should we also replace the help message in the `pre_save_hook` trait with something like "DEPRECATED, use `register_pre_save_hook() instead`"?

Reply:
That's a good question. 

I guess I was imagining we could support both the `register_pre_save_hook` API (for extensions) and the `pre_save_hook` trait (for users). The `pre_save_hook` can only add a single function, provided by the user config system, to the `_pre_save_hooks` list. It will likely get added first in line. Then anything added by the `register_pre_save_hook` API comes next in the list.

What are your thoughts?

Reply:
That's what I implemented so far (but added last, not first).
I don't quite get the users/extensions difference though. An extension can set the `pre_save_hook` trait too, right? Maybe we should add a warning in its validation method, that would be shown if the hook is overriding an existing one?

Reply:
> An extension can set the pre_save_hook trait too, right?

Yes—meaning there's nothing explicitly blocking an extension from setting this trait (and thus, overwriting the user's configured value). But I think this practice should be discouraged, especially from core extensions like JupyterLab, since it's not transparent to the user when an extension is replacing their configuration.

> Maybe we should add a warning in its validation method, that would be shown if the hook is overriding an existing one?

This is a great idea! I think so.